src/main/java/core/basesyntax/dao/FruitTransactionDaoImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.DataBase;
+
+public class FruitTransactionDaoImpl implements FruitTransactionDao{
+    @Override
+    public void add(String fruit, int quantity) {
+        DataBase.storage.put(fruit, quantity);
```

Comment:
The 'add' method should check if the fruit already exists in the storage and increase the quantity instead of setting a new value. This current approach will overwrite any existing quantity.